### ISSUE TITLE
SQL Injection

### ISSUE THREAD

**Issue by pavanagrawal123** [xrayhunter/Python-PostgreSQL-Connection-Demo#1]
Hi!

Just wanted to leave an FYI: 
over here
![image](https://user-images.githubusercontent.com/17802917/47627549-b51e2480-dafe-11e8-8496-7a5b2e107bbf.png)
and 
![image](https://user-images.githubusercontent.com/17802917/47627554-ba7b6f00-dafe-11e8-8069-6523ecec8cbb.png)
you're open to SQL injection because you're passing in raw values into a query. Psycopg2 has a inbuilt way to avoid this BTW :)

**Comment by xrayhunter** [comment]
Great Thanks!

I'll start working on a hot fix for that soon as I can.

**Comment by pavanagrawal123** [comment]
No problem!

If you don't want to fix it, it's no issue, I just wanted to let you know for the future.

**Comment by xrayhunter** [comment]
Nah, I'm going to be using this project to addition my portfolio on security. Just my question is how can I fix the issue on using psycopg2's execute function. 
I'm getting:
![capture](https://user-images.githubusercontent.com/3002612/47629189-f6123b00-daf6-11e8-8dfe-597c23f69ef9.PNG)
![capture](https://user-images.githubusercontent.com/3002612/47629198-07f3de00-daf7-11e8-9e0b-b5029233d53d.PNG)

I'm noticing that the prepared statement is putting single quotes around my %s. Is there a better solution?

**Comment by pavanagrawal123** [comment]
Just researched a bit and it's not suggested *ever* to parameterize a table name, but if you need to, you can do it like described here; https://stackoverflow.com/questions/13793399/passing-table-name-as-a-parameter-in-psycopg2

also, would highly suggest reading the info here: http://initd.org/psycopg/docs/sql.html

**Comment by xrayhunter** [comment]
Thanks man!

I'm right now writing a possible solution for this.
